File: go/bob/bob.go
package bob

import "strings"
import "unicode"

func anyLetter(s string) bool {
	for _, r := range s {
		if unicode.IsLetter(r) {
			return true
		}
	}
	return false
}

// Hey - Whatever :)
func Hey(remark string) string {
	remark = strings.TrimSpace(remark)
	if len(remark) == 0 {
		return "Fine. Be that way!"
	} else if anyLetter(remark) && strings.ToUpper(remark) == remark {
		if remark[len(remark)-1:] == "?" {
			return "Calm down, I know what I'm doing!"
		}
		return "Whoa, chill out!"
	} else if remark[len(remark)-1:] == "?" {
		return "Sure."
	} else {
		return "Whatever."
	}
}

File: python/bob/bob.py
def response(hey_bob):
    hey_bob = hey_bob.strip()
    if len(hey_bob) == 0:
        return "Fine. Be that way!"
    elif hey_bob.isupper() and hey_bob.isalpha():
        if hey_bob[-1] == "?":
            return "Calm down, I know what I'm doing!"
        else:
            return "Whoa, chill out!"
    elif hey_bob[-1] == "?":
        return "Sure."
    else:
        return "Whatever."
